testbed/tests/testbed.py

```diff
@@ -15,9 +15,13 @@
 
 def run_tests(app, cov, args, report_coverage, run_slow, running_in_ci):
     try:
-        # Wait for the app's main window to be visible.
+        # Wait for the app's main window to be visible. Retrieving the actual main window
+        # will raise an exception until the app is actually initialized, so we check the
+        # underlying variable.
         print("Waiting for app to be ready for testing... ", end="", flush=True)
-        while app.main_window is None or not app.main_window.visible:
+        while app._main_window is app._UNDEFINED:
+            time.sleep(0.05)
```

Comment:
Instead of making the testbed depend on these internal symbols, we can just catch the exception.

Reply:
Also, both of these loops should have a time limit, and print a message if it's exceeded. Otherwise, if one of them fails to complete, the tests will hang forever and give no indication of why.